controllers/object_controls.go

```diff
@@ -1539,6 +1543,55 @@ func TransformSandboxDevicePlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPo
 	return nil
 }
 
+// TransformDRADriverPlugin transforms nvidia-dra-driver-plugin daemonset with required config as per ClusterPolicy
+func TransformDRADriverPlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n ClusterPolicyController) error {
+	// update validation container
+	err := transformValidationInitContainer(obj, config)
+	if err != nil {
+		return err
+	}
+
+	// update image
+	image, err := gpuv1.ImagePath(&config.DRADriver)
+	if err != nil {
+		return err
+	}
+	obj.Spec.Template.Spec.Containers[0].Image = image
+
+	// update image pull policy
+	obj.Spec.Template.Spec.Containers[0].ImagePullPolicy = gpuv1.ImagePullPolicy(config.DRADriver.ImagePullPolicy)
+
+	// set image pull secrets
+	if len(config.DRADriver.ImagePullSecrets) > 0 {
+		addPullSecrets(&obj.Spec.Template.Spec, config.DRADriver.ImagePullSecrets)
+	}
+
+	// set resource limits
+	if config.DRADriver.Resources != nil {
+		// apply resource limits to all containers
+		for i := range obj.Spec.Template.Spec.Containers {
+			obj.Spec.Template.Spec.Containers[i].Resources.Requests = config.DRADriver.Resources.Requests
+			obj.Spec.Template.Spec.Containers[i].Resources.Limits = config.DRADriver.Resources.Limits
+		}
+	}
+	// set arguments if specified for device-plugin container
+	if len(config.DRADriver.Args) > 0 {
+		obj.Spec.Template.Spec.Containers[0].Args = config.DRADriver.Args
+	}
+	// set/append environment variables for device-plugin container
+	if len(config.DRADriver.Env) > 0 {
+		for _, env := range config.DRADriver.Env {
+			setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), env.Name, env.Value)
+		}
+	}
+
+	if config.Toolkit.IsEnabled() {
+		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), NvidiaCTKPathEnvName, filepath.Join(config.Toolkit.InstallDir, "toolkit/nvidia-ctk"))
```

Comment:
Let's rather update the driver to also use the `nvidia-cdi-hook` path. The following should be sufficient:
```suggestion
		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), NvidiaCTKPathEnvName, filepath.Join(config.Toolkit.InstallDir, "toolkit/nvidia-cdi-hook"))
```

Reply:
Created https://github.com/NVIDIA/k8s-dra-driver/issues/210 as a follow-up.